### PR TITLE
Alert filter

### PR DESCRIPTION
**Pull request by Hunter27**
Created filter functionly for alerts table - segmentId, SenorType+SensorID and severity

### CODE REVIEW

File: WaterLog_Backend/Controllers/SegmentEventsController.cs [hunk 1]
```diff
@@ -86,59 +86,310 @@ public async Task<List<GetAlerts>> GetAlertsByPage(int id = 1)
...
-                            totalSystemLitres,
-                            entry.ResolvedStatus
-                         )
-                    );
+                    ListOfAlerts.Add
+                     (
+                         new GetAlerts
+                         (
+                             entry.OriginalTimeStamp,
+                             (entry.LatestTimeStamp.Subtract(entry.OriginalTimeStamp) < TimeSpan.Zero ? TimeSpan.Zero : entry.LatestTimeStamp.Subtract(entry.OriginalTimeStamp)),
```
Comment (FredJEhlers) [inline WaterLog_Backend/Controllers/SegmentEventsController.cs]: Split over multiple lines

File: WaterLog_Backend/Controllers/SegmentEventsController.cs [hunk 2]
```diff
@@ -86,59 +86,310 @@ public async Task<List<GetAlerts>> GetAlertsByPage(int id = 1)
...
                         var sensorInfo = await _db.Monitors.Where(a => a.Id == entry.SensorId).FirstOrDefaultAsync();
                         var latestReading = await _db.Readings.Where(a => a.MonitorsId == entry.SensorId)
                             .OrderByDescending(a => a.TimesStamp).FirstOrDefaultAsync();
 
+                        ListOfAlerts.Add
+                        (
+                            new GetAlerts
+                            (
+                                entry.FaultDate,
+                                (entry.AttendedDate.Subtract(entry.AttendedDate) < TimeSpan.Zero ? TimeSpan.Zero : entry.FaultDate.Subtract(entry.AttendedDate)),
```
Comment (FredJEhlers) [inline WaterLog_Backend/Controllers/SegmentEventsController.cs]: Split over multiple lines

File: WaterLog_Backend/Controllers/SegmentEventsController.cs [hunk 3]
```diff
@@ -86,59 +86,310 @@ public async Task<List<GetAlerts>> GetAlertsByPage(int id = 1)
...
+                                    perhourwastagelitre = await proc.CalculatePerHourWastageLitre(entry),
+                                    cost = await proc.CalculatePerHourWastageCost(entry);
+
+                            //Find Litre Usage
+                            ListOfAlerts.Add
+                             (
+                                 new GetAlerts
+                                 (
+                                     entry.OriginalTimeStamp,
+                                     (entry.LatestTimeStamp.Subtract(entry.OriginalTimeStamp) < TimeSpan.Zero ? TimeSpan.Zero : entry.LatestTimeStamp.Subtract(entry.OriginalTimeStamp)),
```
Comment (FredJEhlers) [inline WaterLog_Backend/Controllers/SegmentEventsController.cs]: Split over multiple lines

File: WaterLog_Backend/Controllers/SegmentEventsController.cs [hunk 4]
```diff
@@ -86,59 +86,310 @@ public async Task<List<GetAlerts>> GetAlertsByPage(int id = 1)
...
```
Comment (FredJEhlers) [inline WaterLog_Backend/Controllers/SegmentEventsController.cs]: Split over multiple lines

File: WaterLog_Backend/Controllers/SegmentEventsController.cs [hunk 5]
```diff
@@ -86,59 +86,310 @@ public async Task<List<GetAlerts>> GetAlertsByPage(int id = 1)
...
```
Comment (FredJEhlers) [inline WaterLog_Backend/Controllers/SegmentEventsController.cs]: Split over multiple lines

File: WaterLog_Backend/Controllers/SegmentEventsController.cs [hunk 6]
```diff
@@ -86,59 +86,310 @@ public async Task<List<GetAlerts>> GetAlertsByPage(int id = 1)
...
```
Comment (FredJEhlers) [inline WaterLog_Backend/Controllers/SegmentEventsController.cs]: Split over multiple lines

File: WaterLog_Backend/Controllers/SegmentEventsController.cs [hunk 7]
```diff
@@ -86,59 +86,310 @@ public async Task<List<GetAlerts>> GetAlertsByPage(int id = 1)
...
+                            .Where(a => a.TankMonitorsId == entry.SensorId)
+                            .OrderByDescending(a => a.TimeStamp).FirstOrDefaultAsync();
+                        if (entry.SensorId == SenseId && (int)entry.SensorType == (int)EnumSensorType.WATER_LEVEL_SENSOR)
+                        {
                             ListOfAlerts.Add
                             (
+                               new GetAlerts
+                               (
+                                   entry.FaultDate,
+                                   (entry.AttendedDate.Subtract(entry.AttendedDate) < TimeSpan.Zero ? TimeSpan.Zero : entry.FaultDate.Subtract(entry.AttendedDate)),
```
Comment (FredJEhlers) [inline WaterLog_Backend/Controllers/SegmentEventsController.cs]: Split over multiple lines

File: WaterLog_Backend/Controllers/SegmentEventsController.cs [hunk 8]
```diff
@@ -216,7 +473,7 @@ public async Task<List<GetAlerts>> GetAlerts()
                                 new GetAlerts
                                 (
                                     entry.FaultDate,
-                                    (entry.FaultDate.Subtract(entry.AttendedDate) < TimeSpan.Zero ? TimeSpan.Zero : entry.FaultDate.Subtract(entry.AttendedDate)),
+                                    (entry.AttendedDate.Subtract(entry.AttendedDate) < TimeSpan.Zero ? TimeSpan.Zero : entry.FaultDate.Subtract(entry.AttendedDate)),
```
Comment (FredJEhlers) [inline WaterLog_Backend/Controllers/SegmentEventsController.cs]: Split over multiple lines

File: WaterLog_Backend/Controllers/SegmentEventsController.cs.orig [hunk 9]
```diff
@@ -0,0 +1,588 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading.Tasks;
+using Microsoft.AspNetCore.Mvc;
+using Microsoft.EntityFrameworkCore;
```
Comment (FredJEhlers) [inline WaterLog_Backend/Controllers/SegmentEventsController.cs.orig]: Remove `.orig`

File: WaterLog_Backend/Models/EnumSensorType.cs [hunk 10]
```diff
@@ -8,5 +8,6 @@ namespace WaterLog_Backend.Models
     public enum EnumSensorType
     {
         WATER_FLOW_SENSOR = 1,
+        WATER_LEVEL_SENSOR = 2
```
Comment (FredJEhlers) [inline WaterLog_Backend/Models/EnumSensorType.cs]: add trailing comma

File: WaterLog_Backend/Models/Filter.cs [hunk 11]
```diff
@@ -0,0 +1,17 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading.Tasks;
+
+namespace WaterLog_Backend.Models
+{
+    public class Filter
+    {
+        //public int id { get; set; }
```
Comment (FredJEhlers) [inline WaterLog_Backend/Models/Filter.cs]: GC

**Review by FredJEhlers — CHANGES_REQUESTED** [review]

File: WaterLog_Backend/Controllers/SegmentEventsController.cs (hunk 1 above)
Comment (Hunter27) [reply WaterLog_Backend/Controllers/SegmentEventsController.cs]: done

File: WaterLog_Backend/Controllers/SegmentEventsController.cs (hunk 2 above)
Comment (Hunter27) [reply WaterLog_Backend/Controllers/SegmentEventsController.cs]: done

File: WaterLog_Backend/Controllers/SegmentEventsController.cs (hunk 3 above)
Comment (Hunter27) [reply WaterLog_Backend/Controllers/SegmentEventsController.cs]: done

File: WaterLog_Backend/Controllers/SegmentEventsController.cs (hunk 5 above)
Comment (Hunter27) [reply WaterLog_Backend/Controllers/SegmentEventsController.cs]: done

File: WaterLog_Backend/Controllers/SegmentEventsController.cs (hunk 7 above)
Comment (Hunter27) [reply WaterLog_Backend/Controllers/SegmentEventsController.cs]: done

File: WaterLog_Backend/Controllers/SegmentEventsController.cs.orig (hunk 9 above)
Comment (Hunter27) [reply WaterLog_Backend/Controllers/SegmentEventsController.cs.orig]: done

File: WaterLog_Backend/Models/EnumSensorType.cs (hunk 10 above)
Comment (Hunter27) [reply WaterLog_Backend/Models/EnumSensorType.cs]: done

File: WaterLog_Backend/Models/Filter.cs (hunk 11 above)
Comment (Hunter27) [reply WaterLog_Backend/Models/Filter.cs]: removed

**Review by FredJEhlers — APPROVED** [review]

**Pull request merged**